### PR TITLE
feat: add support for ReadRel.LocalFiles

### PR DESCRIPTION
**Pull request by lidavidm**

### CODE REVIEW

**Comment by lidavidm** [comment]
Mostly a mechanical binding of the current Substrait proto. Helper classes were added to reflect the oneofs.

File: ibis_substrait/expr/operations/relations.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+@validator  # type: ignore[misc]
+def valid_uri(arg: str, **kwargs: typing.Dict[typing.Any, typing.Any]) -> str:
+    arg = rlz.instance_of(str, arg)
+    uri = urllib.parse.urlparse(arg)
+    if not uri.scheme:
+        raise ValueError(f"Missing URI scheme in '{arg}'")
+    return arg
+
+
+class FileFormat(Annotable, Comparable):
```
Comment (kszucs) [inline ibis_substrait/expr/operations/relations.py:19]: If it is intented to be an ibis operation, then subclass from `ibis.expr.operations.Node`

File: ibis_substrait/expr/operations/relations.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    """
+    Base class representing the format of files in a LocalFiles.
+    """
+
+    def __equals__(self, other: typing.Any) -> bool:
```
Comment (kszucs) [inline ibis_substrait/expr/operations/relations.py:24]: If you subclass from `Node` then you can remove this method.

File: ibis_substrait/expr/operations/relations.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+import urllib.parse
+
+import ibis.expr.operations.relations as rel
+import ibis.expr.rules as rlz
+from ibis.common.grounds import Annotable, Comparable
+from ibis.common.validators import validator
+
+
+@validator  # type: ignore[misc]
+def valid_uri(arg: str, **kwargs: typing.Dict[typing.Any, typing.Any]) -> str:
```
Comment (kszucs) [inline ibis_substrait/expr/operations/relations.py:11]: Simply call it `uri`?

**Comment by kszucs** [comment]
The translation looks good, but what's the use case of these operations?

**Comment by lidavidm** [comment]
The use case is to be able to represent a query over files local to the execution backend (though, a "file" may very well be something in cloud storage), so that we can support things like Acero/Substrait where there isn't really a table catalog.

We talked offline a bit and I'll rework this to be 1) not Substrait-specific and 2) part of the Ibis core instead of out here.

**Comment by lidavidm** [comment]
I opened https://github.com/ibis-project/ibis/pull/4293

**Pull request closed without merging**